Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,175 @@
+ including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell
+ 5.1 with .NET Framework provides. There is currently no plan by Windows or the .NET team to ship
+ newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+ Windows has a 5+5 year support lifecycle. .NET supports up to 3 years for Long Term Support (LTS)
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new command in Windows that would:
+
+- Be locally discoverable
+- Perform a simple download and install of the latest LTS release of PowerShell
+- Include parameter options to select Stable, Preview, or LTS releases
+
+## Goals/Non-Goals
+
+Goals:
+
+- By default, download and install the latest LTS release to the users CurrentUser Scope
+  - CurrentUser - default installation scope location $env:LOCALAPPDATA\Microsoft\pwsh. If the path
+    exists, remove and replace.
+  - AllUsers - Administrative users may explicit select this scope. AllUsers requires administrative
+    elevation and performs a silent install using the MSI defaults.
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify channel Stable/Preview/LTS
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- User selected versions prior to current (No version parameter)
+- Disconnected scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+- Cross platform support
+
+## Specification
+
+Command name: install-powershell7
+
+The default behavior is to install to CurrentUser scope, with the latest LTS
+release of PowerShell. If the user has administrative
+privilege, they can choose to install to the scope of AllUsers.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\pwsh
+  - If this path already exists, remove and replace.
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers.
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Scope - overrides the default of CurrentUser and allows the user to specify the
+  installation location AllUsers.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or LTS release.
+
+## Syntax
+
+The command `install-powershell7` is lower case and supports `-/--` switch options to be consistent
+with other command-line tools.
+
+```syntax
+install-powershell7  [-channel <option>] [-scope <option>]
+```
+
+The `--channel <option>` must be one of the following:
```

Comment:
I don't like `--channel` for the name of this parameter. It can get confused with `$env:POWERSHELL_DISTRIBUTION_CHANNEL`. I prefer `--release`.

Reply:
Thanks @sdwheeler - Will discuss with Committee.

Reply:
There is some precedence here with using channel for release version - https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script#options 

though the options for their channel are: STS, LTS, 8.0, 7.0, 6.0 etc.

they also have `quality` but that to me sounds incorrect for our usecase.

Reply:
`--release` suggests you can specify a release version to install that specific release version.
We do call `LTS`, `Stable`, `Preview` channels in our doc, for example, in https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_update_notifications?view=powershell-7.3#long-description:
> The following endpoints are currently used for determining the latest version of **each channel**:

Reply:
The committee has agreed on "Channel"